[WIP] Fix Multiple Sheet Exports With Views

mattdfloyd opened the pull request:
* [x] Checked the codebase to ensure that your feature doesn't already exist.
* [x] Checked the pull requests to ensure that another person hasn't already submitted the feature or fix.
* [ ] Adjusted the Documentation.
* [x] Added tests to ensure against regression.

### Description of the Change

This is a work in progress and feedback is appreciated. My assumption is that `activeSheetIndex` is not being increased when a view is used to compose a sheet. 

### Why Should This Be Added?

Fixes #1584

mattdfloyd committed:
Add test that exhibits current behavior

mattdfloyd committed:
Correct test for desired behavior

patrickbrouwers commented:
Thanks for starting a PR! Haven't been able to look into the problem yet. Thanks for already adding the failing tests, that's very valuable!

mattdfloyd committed:
Add test for multiple sheets

Only tests collections and views. A test for fromQuery and mixed export types would probably be beneficial.

mattdfloyd committed:
Bug fix?

mattdfloyd commented:
There's a good chance that an opinionated assumption is made with this change - that when adding a sheet from a view, it should always be added to the last page in the worksheet. But I think that's a safe bet with the current implementation.

patrickbrouwers commented on src/Sheet.php:
```diff
@@ -133,8 +133,10 @@ public function fromView(FromView $sheetExport)
         /** @var Html $reader */
         $reader = IOFactory::createReader('Html');
-        $reader->setSheetIndex($spreadsheet->getActiveSheetIndex());
+        // dump($spreadsheet->getActiveSheetIndex()); // Implementation bug: always return 0
```
Can you remove this dump?

patrickbrouwers commented on src/Sheet.php:
```diff
@@ -133,8 +133,10 @@ public function fromView(FromView $sheetExport)
         $reader = IOFactory::createReader('Html');
-        $reader->setSheetIndex($spreadsheet->getActiveSheetIndex());
+        // dump($spreadsheet->getActiveSheetIndex()); // Implementation bug: always return 0
+        $reader->setSheetIndex($spreadsheet->getSheetCount() - 1); // Insert content into the last sheet
```
Can you add the comment above the line, instead of next to it? (Just our CS preference)

patrickbrouwers commented on src/Sheet.php:
```diff
@@ -133,8 +133,10 @@ public function fromView(FromView $sheetExport)
+        // dump($spreadsheet->getActiveSheetIndex()); // Implementation bug: always return 0
+        $reader->setSheetIndex($spreadsheet->getSheetCount() - 1); // Insert content into the last sheet
         $reader->loadIntoExisting($tempFile, $spreadsheet);
+        $reader->setSheetIndex(0); // return to the first sheet as active
```
Same here about the comment :)

mattdfloyd replied:
I didn't see that this actually did anything in use. Removed but there could be a case in the future where this is needed. The intention was that the generated file would focus the first sheet when opened but it did not.

patrickbrouwers commented:
Looking good, can you have a look at my comments?

mattdfloyd committed:
Clean up

mattdfloyd committed:
StyleCI

mattdfloyd committed:
StyleCI
